src/main/java/com/example/gradecalculator/service/SubjectService.java

```diff
@@ -0,0 +1,153 @@
+package com.example.gradecalculator.service;
+
+import com.example.gradecalculator.entities.SchoolYear;
+import com.example.gradecalculator.entities.Subject;
+import com.example.gradecalculator.entities.User;
+import com.example.gradecalculator.entities.UserSubject;
+import com.example.gradecalculator.model.SubjectTO;
+import com.example.gradecalculator.model.UserSubjectTO;
+import com.example.gradecalculator.repository.SchoolYearRepository;
+import com.example.gradecalculator.repository.SubjectRepository;
+import com.example.gradecalculator.repository.UserRepository;
+import com.example.gradecalculator.repository.UserSubjectRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class SubjectService {
+
+    private final UserSubjectRepository userSubjectRepository;
+    private final SubjectRepository subjectRepository;
+    private final UserRepository userRepository;
+    private final Map<Integer, Set<String>> selectedSubjectsByYear;
```

Comment:
this should be removed! also 'selectSubjectForYear', 'getSelectedSubjectsForYear' and 'removeSubjectForYear'  are not used in the project beside tests